src/prefect/environments/execution/base.py

```diff
@@ -1,41 +1,54 @@
-import json
-from typing import TYPE_CHECKING
+"""
+Environments are JSON-serializable objects that fully describe how to run a flow. Serialization
+schemas are contained in `prefect.serialization.environment.py`.
 
-import prefect
+Different Environment objects correspond to different computation environments -- currently
+the only allowed environment is a `CloudEnvironment`. This is subject to change in the very
+near future.
+
+Environments that are written on top of a type of infrastructure also define how to
+set up and execute that environment. e.g. the `CloudEnvironment` is an environment which
+runs a flow on Kubernetes using a Dask cluster.
 
-if TYPE_CHECKING:
-    import prefect.engine.state
+Some of the information that the environment requires to run a flow -- such as the flow
+itself -- may not available when the Environment class is instantiated. Therefore, Environments
+are accompanied with a Storage objects to specify how and where the flow is stored. For example,
+the `CloudEnvironment` requires the flow to be stored in a `Docker` storage object.
+"""
+
+import prefect
 
 
 class Environment:
     """
+    Base class for Environments.
+
+    An environment is an object that can be instantiated in a way that makes it possible to
+    call `environment.setup()` to stand up any required static infrastructure and
+    `environment.execute()` to execute the flow inside this environment.
+
+    The `setup` and `execute` functions of an environment require a Prefect Storage object which
+    specifies how and where the flow is stored.
     """
 
     def __init__(self) -> None:
         pass
 
-    def process(self, storage: "prefect.environments.storage.Storage") -> None:
-        """"""
-        pass
-
-    def execute(self) -> None:
-        """
-        Executes the environment on any infrastructure created during setup
+    def setup(self, storage: "prefect.environments.storage.Storage") -> None:
```

Comment:
Oh and question hereL: when will `setup` require the storage to be provided? just curious.

Reply:
If I have a setup that wants to spin up preexisting dask workers using the flow's stored image. Is one example